### PR TITLE
feat: added     resolveOnNavigated option

### PR DESCRIPTION
**Pull request by farfromrefug**
# Conflicts:
#	index.d.ts

### CODE REVIEW

**Comment by rigor789** [comment]
Can you describe the use-case for this change?

When would you only want to resolve after the page has been `navigatedFrom`?

**Comment by farfromrefug** [comment]
Actually i needed this because of a bug in {N} where unwanted taps were still sent after starting a navigation so i did something like that:
```
this.shouldIgnoreTaps = true;
                                        // this.hideLoading();
                                        return this.$getAppComponent()
                                            .navigateTo(SessionDetails, { props: { session, startInEdit: true }, resolveOnNavigated: true })
                                            .then(() => {
                                                // setTimeout(() => {
                                                this.shouldIgnoreTaps = false;
                                                // }, 500);
                                            });
```
Before that i needed to do the timeout thing that you see.
I think it can get useful to anyone

**Comment by rigor789** [comment]
Shouldn't that be in a `navigatedTo` event though?

**Comment by farfromrefug** [comment]
@rigor789 you actually right. It was working in my case even though it was wrongly implemented.
So i rewrote it to be really useful.
It is now a`resolveOnEvent` property which can be set to `navigatedTo` or `navigatingTo`

File: index.d.ts [hunk 1]
```diff
@@ -8,7 +8,8 @@ export type NativeScriptVueItemEventData<T> = ItemEventData & { item: T }
 
 export interface NavigationEntryVue extends NavigationEntry {
     props?: Record<string, any>,
-    frame?: any
+    frame?: any,
+    resolveOnNavigated?: Page.navigatingToEvent | Page.navigatedToEvent
```
Comment (rigor789) [inline index.d.ts:12]:
```suggestion
    resolveOnEvent?: Page.navigatingToEvent | Page.navigatedToEvent
```

File: platform/nativescript/plugins/navigator-plugin.js [hunk 2]
```diff
@@ -89,6 +89,10 @@ export default {
 
         updateDevtools()
 
+        const resolveOnEvent = options.resolveOnEvent === true
```
Comment (rigor789) [inline platform/nativescript/plugins/navigator-plugin.js:92]: Wouldn't this make `resolveOnEvent` into a boolean? And then on line 112: 
```js
 page.on(resolveOnEvent, resolveHandler)
```
Would try to do 
```js
 page.on(true, resolveHandler)
```

```suggestion
        const resolveOnEvent = options.resolveOnEvent
```

**Pull request merged**